samples/lorawan/class_a/src/main.c

```diff
@@ -25,13 +25,32 @@ BUILD_ASSERT(DT_NODE_HAS_STATUS(DEFAULT_RADIO_NODE, okay),
 					  0x09, 0xCF, 0x4F, 0x3C }
 
 #define DELAY K_MSEC(10000)
+#define DL_MSG_LEN 256
 
 #define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
 #include <logging/log.h>
 LOG_MODULE_REGISTER(lorawan_class_a);
 
 char data[] = {'h', 'e', 'l', 'l', 'o', 'w', 'o', 'r', 'l', 'd'};
 
+void downlink_printer(uint8_t port, uint8_t length, uint8_t *payload)
+{
+	static char hex_str[DL_MSG_LEN*2+1];
+	int i;
+
+	for (i = 0; i < length; i++) {
+		hex_str[i*2] = (payload[i] & 0xF0) >> 4;
+		hex_str[i*2] = hex_str[i*2] < 10 ?
+			'0' + hex_str[i*2] : 'A' + hex_str[i*2];
+		hex_str[i*2+1] = payload[i] & 0x0F;
+		hex_str[i*2+1] = hex_str[i*2+1] < 10 ?
+			'0' + hex_str[i*2+1] : 'A' + hex_str[i*2+1] - 10;
```

Comment:
Can't you use `bin2hex`?

Reply:
Didn't realise that util existed. Using it now.